### PR TITLE
POC build during the build phase.

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

**Comment by stefanpenner** [comment]
This is intended as  POC, I would like to recommend the html-next team take this direction for the build steps.

cc @runspired

**Comment by rwwagner90** [comment]
I believe @runspired was going to help us through how to get this over the finish line

**Comment by kybishop** [comment]
I'm pretty sure I understand what's going on here. Digging into this now

File: lib/scss-variables-compiler.js [hunk 1]
```diff
@@ -13,14 +14,28 @@ function makeBreakpointLine(bp, isLast) {
...
+    this.options = options;
 
-  // clean out the old _variables.scss
-  if (FS.existsSync(filePath)) {
-    FS.unlinkSync(filePath);
+    this._hasBuilt = false;
   }
 
+  build() {
+    if (this._hasBuilt) { return; }
```
Comment (kybishop) [inline lib/scss-variables-compiler.js]: @stefanpenner For my own education.. how might we end up calling `build()` a second time after the first call to `build()`? Also, much thanks for this PR, it has been very educational on broccoli as is!

**Comment by kybishop** [comment]
Bringing this across the finish line in #9. Thanks a ton @stefanpenner!

File: index.js [hunk 2]
```diff
@@ -37,13 +38,19 @@ module.exports = {
...
-
     this.app = app;
     return app;
   },
 
+  treeForAddonStyles(tree) {
+    console.log('OMG')
+    let stylesRoot = path.join(__dirname, 'addon/styles');
+    return mergeTrees([
+      stylesRoot,
```
Comment (kybishop) [inline index.js]: @stefanpenner Is there a particular reason you went with `path.join(__dirname, 'addon/styles')` instead of just using `tree`?

**Pull request closed without merging**